### PR TITLE
Use require.resolve to find CSS modules in node_modules

### LINKED ISSUE #136: Ctrl+click (go to file) doesn't work for saas-loader ~ imports (node_modules)
Trying to Ctrl+click on the imported module name fails:
```scss
@import '~bootstrap/scss/bootstrap';
```
with this error:

```
Unable to open '_bootstrap.scss': File not found (file:///PROJECT/src/styles/~bootstrap/scss/_bootstrap.scss).
```

The reason is that I'm using `webpack` with `saas-loader`, and `saas-loader` treats an import path starting with `~` as an import from `node_modules`: https://github.com/webpack-contrib/sass-loader#imports

So the actual location from where the import should be loaded is:
```
file:///PROJECT/node_modules/bootstrap/scss/_bootstrap.scss
```

### LINKED ISSUE #78894: [scss] link doesn't work with ~Tilde to search in node module or webpack alias
Using a **~**Tilde to go directly to a node_module path without having to add aliases to webpack this path doesn't get resolved by VS Code. 

So with a node_module 'bootstrap' installed, we can import bootstrap sass-files using this construction:

``` scss
@import '~bootstrap/scss/bootstrap-grid';
@import '~bootstrap/scss/utilities/sizing';
```
While this is working fine with Webpack and the sass-loader/node_sass, VS Code doesn't recognise this tilde-construction and searches on a follow-link in our `source-folder\bootstrap\` instead of in `node_modules\bootstrap\`

When we use aliases in Javascript we solved his resolve problem by adding aliases to the jsconfig.json. But AFAIK there's not such a config file for Sass in VS Code.
I've been searching everywhere, including settings on sass, but couldn't find anything where we could make VS Code respect the node_modules path (and Webpack aliases) with a tilde~.

So unfortunately we seem to not being able to click to follow link on these sass imports.

**versions**
- VSCode Version: 1.37.0
- OS Version: Windows 10 Home Updated

**Steps to Reproduce:**
1. Have a node modulle, like node_modules/bootstrap, installed
2. Import a sassfile from this module, like `@import '~bootstrap/scss/bootstrap-grid';`
3. Ctrl+click on the import line to try to 'Follow Link'
Result: it shows an error-message popup saying 'Unable to open' with the wrong path

**extensions**
Does this issue occur when all extensions are disabled?: Yes

### PR DESCRIPTION
**Pull request by penx**
Further to #70693 which was reworked by @aeschli  in https://github.com/microsoft/vscode/commit/8f7293470661319099bdda43eecc386a43c87189 due to issues with require.resolve and webpack.

Fixes #78894
Fixes microsoft/vscode-css-languageservice#136

The current implementation of resolving the path to a CSS module in node_modules has significant limitations.

Using require.resolve directly should be the most solid approach but can't be used in code that is webpacked.

This PR allows us to use require.resolve via a module proxy ([requireresolveproxy](https://www.npmjs.com/package/requireresolveproxy)) and marking the module as a commonjs external in the webpack config.

Potential alternative approaches:

- implement as per https://github.com/microsoft/TypeScript/blob/master/src/compiler/moduleNameResolver.ts (ideally by importing it directly and with support from someone who wrote this)
- use https://github.com/browserify/resolve
- use require.resolve in the same way as it's used here https://github.com/microsoft/vscode-languageserver-node/blob/master/server/src/resolve.ts (ideally with support/direction from someone who knows how we could replicate this in css-language-features, @dbaeumer?)

Screenshot 2019-08-23 at 16 18 58
Screenshot 2019-08-23 at 16 19 48

### CODE REVIEW

**Comment by penx** [comment]
This builds fine now when I run `yarn gulp -- vscode-darwin`, however I'm not sure how I run the production build on MacOS in order to test it. I have tried `cd out-build && node main.js` but get `Error: Cannot find module 'electron'`.

**Comment by penx** [comment]
Confirmed this is working in the dev build (`yarn watch` and `./scripts/code.sh`) so taking out of draft - I can test in the production build too if someone lets me know how!

File: extensions/css-language-features/server/src/utils/documentContext.ts [hunk 1]
```diff
@@ -8,9 +8,10 @@ import { endsWith, startsWith } from '../utils/strings';
 import * as url from 'url';
 import { WorkspaceFolder } from 'vscode-languageserver';
 import URI from 'vscode-uri';
-import { join, dirname } from 'path';
 import { existsSync } from 'fs';
 
+const resolve = require('requireresolveproxy');
```
Comment (penx) [inline extensions/css-language-features/server/src/utils/documentContext.ts]: this is due to not having .ts typings for requireresolveproxy - happy to resolve in another way if someone can suggest how

**Comment by penx** [comment]
Pretty sure the failing CI is unrelated to this PR

```
  49 passing (2s)
  1 failing

  1) Links node module resolving:

      AssertionError [ERR_ASSERTION]: 'file:///d%3A/a/1/s/extensions/css-language-features/server/test/linksTestFixtures/foo/hello.html' == 'file:///d%3A/a/1/s/extensions/css-language-features/server/test/linksTestFixtures/node_modules/foo/hello.html'
      + expected - actual

      -file:///d%3A/a/1/s/extensions/css-language-features/server/test/linksTestFixtures/foo/hello.html
      +file:///d%3A/a/1/s/extensions/css-language-features/server/test/linksTestFixtures/node_modules/foo/hello.html
      
      at assertLink (D:\a\1\s\extensions\css-language-features\server\out\test\links.test.js:23:16)
      at assertLinks (D:\a\1\s\extensions\css-language-features\server\out\test\links.test.js:37:13)
```

**Comment by aeschli** [comment]
That's the test that I added for the node module resolving.

**Comment by penx** [comment]
Replaced by #81555

**Pull request closed without merging**